src/resource/task/task.controller.ts

```diff
@@ -0,0 +1,82 @@
+import {
+    Body,
+    Controller,
+    Get,
+    HttpStatus,
+    Param,
+    ParseIntPipe,
+    Patch,
+    UseGuards,
+    Post,
+    Delete,
+    Res,
+} from '@nestjs/common';
+import { GetUser } from 'src/decorators';
+import { JwtGuard } from 'src/auth/guard';
+import { EditTaskDTO, ReturnTaskDTO, CreateTaskDTO } from './dto';
+import { TaskService } from './task.service';
+import { Response } from 'express';
+
+@UseGuards(JwtGuard)
+@Controller('tasks')
+export class TaskController {
+    constructor(private readonly taskService: TaskService) {}
+
+    @Get(':id')
+    async getTask(
+      @GetUser('id') userId: number,
+      @Param('id', ParseIntPipe) taskId: number,
+      @Res() res: Response,
+    ): Promise<void> {
+      const task = await this.taskService.getTask(userId, taskId);
+      if(task){
+        res.status(HttpStatus.OK).json(task);
+      }else{
+        res.status(HttpStatus.NOT_FOUND).json({ message: 'Task not found' });
+      }
+    }
+
+    @Get()
+    getAllTasks(
```

Comment:
This can be good at the future, but you missed the most important endpoint - fetching all user tasks.